services/web/server/src/simcore_service_webserver/director_v2/_handlers.py

```diff
@@ -98,18 +100,41 @@ async def start_computation(request: web.Request) -> web.Response:
 
     # Get wallet information
     wallet_info = None
-    project_wallet = await projects_api.get_project_wallet(
-        request.app, project_id=project_id
-    )
+    product = products_api.get_current_product(request)
     app_settings = get_settings(request.app)
-    if project_wallet and app_settings.WEBSERVER_CREDIT_COMPUTATION_ENABLED:
-        # Check whether user has access to the wallet
-        await wallets_api.get_wallet_by_user(
-            request.app, req_ctx.user_id, project_wallet.wallet_id, req_ctx.product_name
+    if product.is_payment_enabled and app_settings.WEBSERVER_CREDIT_COMPUTATION_ENABLED:
+        project_wallet = await projects_api.get_project_wallet(
+            request.app, project_id=project_id
         )
-        wallet_info = WalletInfo(
-            wallet_id=project_wallet.wallet_id, wallet_name=project_wallet.name
+        if project_wallet is None:
+            user_default_wallet_preference = await user_preferences_api.get_user_preference(
+                request.app,
+                user_id=req_ctx.user_id,
+                product_name=req_ctx.product_name,
+                preference_class=user_preferences_api.PreferredWalletIdFrontendUserPreference,
+            )
+            if user_default_wallet_preference is None:
+                raise ValueError(
```

Comment:
Please either extend the `DirectorServiceError` and create a new error or reuse that one.

Reply:
thanks i forgot, it is done :+1:

Reply:
Ups by mistake i merged before pushing the change, I will add it in the next PR on Monday. Sorry for that.